nsxt/resource_nsxt_policy_mac_discovery_profile.go

```diff
@@ -14,13 +15,116 @@ import (
 
 	"github.com/vmware/terraform-provider-nsxt/api/infra"
 	utl "github.com/vmware/terraform-provider-nsxt/api/utl"
+	"github.com/vmware/terraform-provider-nsxt/nsxt/metadata"
 )
 
 var macDiscoveryProfileMacLimitPolicyValues = []string{
 	model.MacDiscoveryProfile_MAC_LIMIT_POLICY_ALLOW,
 	model.MacDiscoveryProfile_MAC_LIMIT_POLICY_DROP,
 }
 
+var macDiscoveryProfileSchema = map[string]*metadata.ExtendedSchema{
+	"nsx_id":       metadata.GetExtendedSchema(getNsxIDSchema()),
+	"path":         metadata.GetExtendedSchema(getPathSchema()),
+	"display_name": metadata.GetExtendedSchema(getDisplayNameSchema()),
+	"description":  metadata.GetExtendedSchema(getDescriptionSchema()),
+	"revision":     metadata.GetExtendedSchema(getRevisionSchema()),
+	"tag":          metadata.GetExtendedSchema(getTagsSchema()),
+	"context":      metadata.GetExtendedSchema(getContextSchema()),
+	"mac_change_enabled": {
+		Schema: schema.Schema{
+			Type:     schema.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "bool",
+			SdkFieldName: "MacChangeEnabled",
+			TestData: metadata.Testdata{
+				CreateValue: "true",
+				UpdateValue: "false",
+			},
+		},
+	},
+	"mac_learning_enabled": {
+		Schema: schema.Schema{
+			Type:     schema.TypeBool,
+			Optional: true,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "bool",
+			SdkFieldName: "MacLearningEnabled",
+			TestData: metadata.Testdata{
+				CreateValue: "true",
+				UpdateValue: "false",
+			},
+		},
+	},
+	"mac_limit": {
+		Schema: schema.Schema{
+			Type:         schema.TypeInt,
+			Optional:     true,
+			ValidateFunc: validation.IntBetween(0, 4096),
+			Default:      4096,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "int",
+			SdkFieldName: "MacLimit",
+			TestData: metadata.Testdata{
+				CreateValue: "20",
+				UpdateValue: "50",
+			},
+		},
+	},
+	"mac_limit_policy": {
+		Schema: schema.Schema{
+			Type:         schema.TypeString,
+			ValidateFunc: validation.StringInSlice(macDiscoveryProfileMacLimitPolicyValues, false),
+			Optional:     true,
+			Default:      "ALLOW",
```

Comment:
I suggest we use the constant here and below

Reply:
done